### PR TITLE
quickgui: init at 1.2.8

### PR DESCRIPTION
**Pull request by heyimnova**
###### Description of changes

Add [quickgui](https://github.com/quickemu-project/quickgui) which is a Flutter frontend for [quickemu](https://github.com/quickemu-project/quickemu), a tool for quickly creating and running optimized virtual machines.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/virtualization/quickgui/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    wrapGAppsHook
+  ];
+
+  unpackCmd = "dpkg-deb -x $curSrc source";
+
+  installPhase = ''
+    runHook preInstall
+
+    mv usr $out
+
```
Comment (AndersonTorres) [inline pkgs/applications/virtualization/quickgui/default.nix:33]:
```suggestion
```

File: pkgs/applications/virtualization/quickgui/default.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      --replace "/usr" $out
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A Flutter frontend for quickemu";
+    homepage = "https://github.com/quickemu-project/quickgui";
+    downloadPage = "https://github.com/quickemu-project/quickgui/releases/tag/v${version}";
+    changelog = "https://github.com/quickemu-project/quickgui/compare/v1.2.7...v1.2.8";
```
Comment (AndersonTorres) [inline pkgs/applications/virtualization/quickgui/default.nix:44]:
```suggestion
    changelog = "https://github.com/quickemu-project/quickgui/releases/tag/v${version}";
```

File: pkgs/applications/virtualization/quickgui/default.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    substituteInPlace $out/share/applications/quickgui.desktop \
+      --replace "/usr" $out
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A Flutter frontend for quickemu";
+    homepage = "https://github.com/quickemu-project/quickgui";
+    downloadPage = "https://github.com/quickemu-project/quickgui/releases/tag/v${version}";
```
Comment (AndersonTorres) [inline pkgs/applications/virtualization/quickgui/default.nix:43]:
```suggestion
```

Not needed indeed.

File: pkgs/applications/virtualization/quickgui/default.nix [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    mv usr $out
+
+    substituteInPlace $out/share/applications/quickgui.desktop \
+      --replace "/usr" $out
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/applications/virtualization/quickgui/default.nix:40]:
```suggestion
  meta = {
```

File: pkgs/applications/virtualization/quickgui/default.nix [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    description = "A Flutter frontend for quickemu";
+    homepage = "https://github.com/quickemu-project/quickgui";
+    downloadPage = "https://github.com/quickemu-project/quickgui/releases/tag/v${version}";
+    changelog = "https://github.com/quickemu-project/quickgui/compare/v1.2.7...v1.2.8";
+    maintainers = [ maintainers.heyimnova ];
+    mainProgram = "quickgui";
+    platforms = platforms.linux;
+    sourceProvenance = [ sourceTypes.binaryNativeCode ];
```
Comment (AndersonTorres) [inline pkgs/applications/virtualization/quickgui/default.nix:48]:
```suggestion
    maintainers = [ lib.maintainers.heyimnova ];
    platforms = lib.platforms.linux;
    sourceProvenance = [ lib.sourceTypes.binaryNativeCode ];
    mainProgram = "quickgui";
```

Comment (AndersonTorres) [reply pkgs/applications/virtualization/quickgui/default.nix:48]: PS.: is it too hard to compile the source?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/virtualization/quickgui/default.nix (hunk 5 above)
Comment (heyimnova) [reply pkgs/applications/virtualization/quickgui/default.nix:47]: I did try to use mkFlutterApp but it didn't work and there seems to be no documentation for building Flutter applications with Nix, I did find two threads in the NixOS forums however neither of them could work out what to do in their own cases.

Comment (AndersonTorres) [reply pkgs/applications/virtualization/quickgui/default.nix]: Good to know. I will backlog this on issues.

**Comment by AndersonTorres** [comment]
Welcome to Nixpkgs! I will be your guide!

In Nixpkgs, the pull requests follow the model below:

- The first commit includes you to the maintainers file. 
   - It must be titled `maintainers: add <your name>`
   - Ideally, you do this once and then forget

- The second commit includes the expression you are writing
  - It should be titled `<appname>: init at <version>`
  - `<appname>` should be all lowercase (there are some exceptions)
  - `<version>` should start with a digit (except when picking a specific Git commit, in which you will use `unstable-YYYY-MM-DD`)

----------

That being said, use `git squash` to make your contribution follow the guide above.

If any doubt, drop me a ping!

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Read the comment above

**Comment by heyimnova** [comment]
> Read the comment above

I have sorted out the commits, it now follows the model above

**Comment by AndersonTorres** [comment]
As far as I understand the docs, it is a GUI to quickemu. What happens when quickemu is not previously installed?

**Comment by heyimnova** [comment]
image
It will show this error screen if it cannot find quickemu

**Comment by AndersonTorres** [comment]
Acceptable. I will merge.

**Pull request merged**